Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp

```diff
@@ -342,6 +367,104 @@ void AudioSystem::updateProjectileLowPass(Audio::SoundInfo_General* general) {
 	m_audioEngine->updateProjectileLowPass(general->frequency, general->soundID);
 }
 
+void AudioSystem::dealWithDeathSound(AudioComponent* audioC, float dt) {
+	Audio::SoundInfo_General* soundGeneral = &audioC->m_sounds[Audio::SoundType::DEATH];
+	if (soundGeneral->isPlaying) {
+		int soundPoolSize = audioData.m_soundsUnique[Audio::SoundType::DEATH].size();
+
+		if (soundPoolSize > 0) {
+			if (soundGeneral->isPlaying) {
```

Comment:
Double check in same nested if-statement

Reply:
Fixed